Tensile/Source/lib/source/ContractionSolution.cpp

```diff
@@ -1228,11 +1228,19 @@ namespace Tensile
 
         if(sizeMapping.streamK >= 2)
         {
-            if(debug)
-                rv.push_back(generateStreamKInitCall<TypedInputs, true>(problem, inputs, hardware));
-            else
-                rv.push_back(
-                    generateStreamKInitCall<TypedInputs, false>(problem, inputs, hardware));
+            auto   tiles  = problem.getNumTiles(sizeMapping);
+            size_t skGrid = getSKGrid(hardware, tiles);
+            // If problem can run full data-parallel, flags are not needed since there is no fixup step
```

Comment:
If we do not need flags in this case, I think it is better to change workspace size calculation as well (not allocate workspace for flag).

Reply:
Good call, posted an update.